src/Uno.Extensions.Core.Generators/PropertySelector/PropertySelectorCandidate.cs

```diff
@@ -1,43 +1,207 @@
-﻿using System;
+﻿using System.Collections.Generic;
+using System.Collections.Immutable;
 using System.Diagnostics.CodeAnalysis;
+using System.Globalization;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading;
 using Microsoft.CodeAnalysis;
 using Microsoft.CodeAnalysis.CSharp;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Uno.Extensions.Core.Generators.Helpers;
+using Uno.Extensions.Edition;
 
 namespace Uno.Extensions.Generators.PropertySelector;
 
 internal readonly record struct PropertySelectorCandidate
```

Comment:
This model has more fields now, and I'm not very sure if `struct` is still good here. @dr1rrb should we switch this to a record class?

Side note: One issue with this generator is that it is triggered by `InvocationExpressionSyntax`, and we have no way to filter them down to few ones syntactically. So in any case, we have to create the model for **every** invocation in the compilation and get semantic info about the model being called for **every** invocation.

The good thing about this generator though is that it uses `RegisterImplementationSourceOutput`, so a good host will not re-run on every typed character. So this is effectively run only on build (can run in niche scenarios like EnC/HotReload when the IDE cares about the code being generated).

Reply:
For the `class` vs `struct`, your call :) I may be wrong, but I've the feeling that some of those properties are no longer used since you generate the code much earlier.

Reply:
SyntaxNode wasn't used indeed. But the rest appears to be used.